lib/Event/ViewDeletedEvent.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OCA\Tables\Event;
+
+use OCA\Tables\Db\View;
+use OCP\EventDispatcher\Event;
+
+final class ViewDeletedEvent extends Event {
+	public function __construct(protected View $view, protected string $userId) {
+		parent::__construct();
+	}
+
+	public function getView(): View {
+		return $this->view;
+	}
+
+	public function getUserId(): string {
+		return $this->userId;
```

Comment:
We could probably leave passing the user id to every event as it can always be obtained from the user session and enhance the log message with it from the AuditLogService

Reply:
Really good take, initially I thought the Event is more general purpose than just logging and always need to wrap all the necessary data because it knows nothing about the Listener classes (Logging logic) and because in the future there’s some other Listeners observing it (not only logging) so they might need those data (userId) but yes we can always get that from global session one (But yeah with some pattern like distributed event driven system, user might not be accessible globally haha)


Definitely agree with you the Logging Listener no need the duplication of userId there because it’s already on the Logging Service

Reply:
Yes, something we can probably consider once we come up with use cases that require that. So far we don't have a global event bus system in Nextcloud nor we do have code paths that would require calling operations in scope of other users within tables, but thanks for bringing up that topic and explaining the background.